plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetastoreModule.java

```diff
@@ -39,4 +67,12 @@ public MetastoreValidator(HiveMetastore metastore)
             }
         }
     }
+
+    private void bindMetastoreModule(String name, Module module)
+    {
+        install(conditionalModule(
+                MetastoreTypeConfig.class,
```

Comment:
We should make this an Iceberg specific config with an enum, so that you can't use the wrong type. But that can be a follow up.